crates/deltalake-core/src/writer/test_utils.rs

```diff
@@ -334,6 +334,21 @@ pub mod datafusion {
             .unwrap()
     }
 
+    pub async fn get_data_sorted(table: &DeltaTable, columns: &str) -> Vec<RecordBatch> {
```

Comment:
You don't need to sort the data yourself and can reuse this utility: https://github.com/delta-io/delta-rs/blob/main/crates/deltalake-core/src/writer/test_utils.rs#L325

Reply:
The reason I added that extra method was that, unlike a straightforward write, after `replaceWhere` `get_data` was returning rows in some mixed order and `assert_batches_sorted_eq` was not able to match them. I could of course set the `expected` to be in the same order as returned by `get_data` but I thought that it could still be vulnerable to some future changes that could for example change the order returned by `get_data`. 

I can remove this method and use `get_data` with `expected` being in a specific order as I am getting today if you think that order will never change.

Reply:
I'm a bit surprised it was failing. The order given by `get_data` should not matter since the assert marco performs a sort on the string representation of the batches. 
Your not supposed to set the `expected` string to whatever `get_data` returns. The order depends on the string representation. When using this macro I typically put some garbage string for `expected`. let the test fail, manually validate the `actual` results (which will be ordered) and then update `expected` with actual.

Reply:
That's exactly what I did to find out what the `actual` was. I'll try again using only `get_data` and `assert_batches_sorted_eq`, maybe I messed something up when I tried before

Reply:
Ahh, I remembered the other reason now, `get_data` was messing up the column order too so I needed a version of `get_data` where I could pass the column names (`get_data` uses `*`). Once I added that function I decided to include `order by` too. Here is what I get in my test if I use `get_data`:
```
expected:

[
    "+----+-------+------------+",
    "| id | value | modified   |",
    "+----+-------+------------+",
    "| A  | 11    | 2024-02-02 |",
    "| A  | 13    | 2024-02-02 |",
    "| A  | 15    | 2024-02-01 |",
    "| B  | 2     | 2021-02-02 |",
    "| B  | 4     | 2021-02-01 |",
    "| B  | 8     | 2021-02-01 |",
    "| B  | 9     | 2021-02-01 |",
    "+----+-------+------------+",
]
actual:

[
    "+------------+----+-------+",
    "| modified   | id | value |",
    "+------------+----+-------+",
    "| 2021-02-01 | B  | 4     |",
    "| 2021-02-01 | B  | 8     |",
    "| 2021-02-01 | B  | 9     |",
    "| 2021-02-02 | B  | 2     |",
    "| 2024-02-01 | A  | 15    |",
    "| 2024-02-02 | A  | 11    |",
    "| 2024-02-02 | A  | 13    |",
    "+------------+----+-------+",
]
```

Reply:
Got it. Yeah this is just how datafusion functions with partition columns. You can keep that utility function to maintain column order.